README.md

```diff
@@ -1,19 +1,92 @@
 # Official VOLTTRON docker image
 
 # Introduction
-This image provides a reproducable way to install VOLTTRON within a docker container.  It features gosu which allows the storage of VOLTTRON_HOME to be persistable on the hosts hard drive.  The
-base docker images are available on docker hub at https://hub.docker.com/r/volttron/volttron/.
+This image provides a reproducable way to install VOLTTRON within a docker container.  It features gosu which allows the storage of VOLTTRON_HOME to be persistable on the hosts hard drive.
+The base docker images are available on docker hub at https://hub.docker.com/r/volttron/volttron/.
 
-# Usage
+# Platform Initialization
 
-```` bash
+The VOLTTRON container when created is just a blank container with no agents.  Now there is an initialization routine available within the docker container to allow the installation of agents before launching of the instance.  To do this one will mount a `platform_config.yml` file to `/platform_config.yml` within the container. One is also likely to need to mount agent configurations (specified in the `platform_config.yml` file), into the container. The recommended way to do this is through a `docker-compose.yml` file.  An example of this is included in this repository, based on the one in the [volttron-fuel-cells repo](https://github.com/VOLTTRON/volttron-fuel-cells/).
+
+The `platform_config.yml` file has two sections: `config`, which configures the main instance and populate's the main config file ($VOLTTRON_HOME/config), and `agents`, which contains a list of agents with references to configurations for them (note the frequent use of environment variables in this section).
+
+## Main Configuration
+The main instance configuration is composed of key value pairs under a "config" key in the `platofrom_config.yml` file.
+As an example, the `vip-address` and `bind-web-address` would be populated using the following partial file:
+``` yaml
+# Properties to be added to the root config file:
+# - the properties should be ingestable for volttron
+# - the values will be presented in the config file
+#   as key=value
+config:
+  vip-address: tcp://0.0.0.0:22916
+  bind-web-address: http://0.0.0.0:8080
+  # volttron-central-address: a different address
+  # volttron-central-serverkey: a different key 
+
+  ...
+```
+
+## Agent Configuration
+The agent configuration section is under a top-level key "agents" and contains several layers of nested key-value mappings.
+The top level of the section is keyed with the names of the desired agents, each of which contains a mapping.
```

Comment:
The "name" of each agent in this file maps onto the identity of the installed agent.  So foo.baz agent would have that identity.  This is important because it is one of the main ways for agents to communicate both within and between instances of volttron.